### PR TITLE
[LaTeXWriter] fix escaping in header titles

### PR DESCRIPTION
**Pull request by odow**
I think this is the culprit of https://github.com/jump-dev/JuMP.jl/pull/3386. I'll test on this branch before adding tests etc.

### CODE REVIEW

**Comment by mortenpi** [comment]
Added a test here that should hopefully fail on current master: https://github.com/JuliaDocs/Documenter.jl/commit/e1396e557829ff96e70a29fdc4b1a32865793d77.

File: src/latex/LaTeXWriter.jl [hunk 1]
```diff
@@ -83,6 +83,7 @@ Context(io, doc) = Context{typeof(io)}(io, false, Dict(), 1, "", doc)
 
 _print(c::Context, args...) = Base.print(c.io, args...)
 _println(c::Context, args...) = Base.println(c.io, args...)
+_print(io, args...) = Base.print(io, args...)
```
Comment (odow) [inline src/latex/LaTeXWriter.jl:86]: I guess the single-argument `latexesc(s)` method wasn't actually tested, because it hit this method error.

**Comment by mortenpi** [comment]
I got the test to finally fail on master, so I took the liberty of pushing it here.

**Review by odow — COMMENTED** [review]
LGTM

**Comment by mortenpi** [comment]
It looks like the JuMP build also passed with this fix?

**Comment by odow** [comment]
Yip. Although now I need to tweak again, because it made every docstring a Chapter 😆

**Comment by mortenpi** [comment]
Let's hope it doesn't reveal more bugs -- I'll go ahead and merge this :laughing:

**Pull request merged**